django/db/models/functions/json.py

```diff
@@ -0,0 +1,196 @@
+from django.db.models.constants import LOOKUP_SEP
+from django.db.models.expressions import Func, Value
+from django.db.models.fields.json import JSONField, compile_json_path
+from django.db.models.functions import Cast
+
+
+class JSONSet(Func):
+    output_field = JSONField()
+
+    def __init__(self, field_name, **fields):
```

Comment:
Since `field_name` doesn't actually have to be a field name (it can be any expression in the current query), I think we should rename this to `expression`

```suggestion
    def __init__(self, expression, **fields):
```